# Test info

- Name: (2) user failed login using wrong_user
- Location: /__w/pw-github-action/pw-github-action/tests/souce-demo.spec.js:51:9

# Error details

```
Error: Timed out 5000ms waiting for expect(locator).toHaveText(expected)

Locator: locator('[data-test=\'error\']')
Expected string: "Epic sadface: Password is required..."
Received string: "Epic sadface: Password is required"
Call log:
  - expect.toHaveText with timeout 5000ms
  - waiting for locator('[data-test=\'error\']')
    9 × locator resolved to <h3 data-test="error">…</h3>
      - unexpected value "Epic sadface: Password is required"

    at /__w/pw-github-action/pw-github-action/tests/souce-demo.spec.js:59:59
```

# Page snapshot

```yaml
- text: Swag Labs
- textbox "Username": wrong_user
- textbox "Password"
- 'heading "Epic sadface: Password is required" [level=3]':
  - button
  - text: "Epic sadface: Password is required"
- button "Login"
- heading "Accepted usernames are:" [level=4]
- text: standard_user locked_out_user problem_user performance_glitch_user error_user visual_user
- heading "Password for all users:" [level=4]
- text: secret_sauce
```

# Test source

```ts
   1 | import { test, expect } from '@playwright/test';
   2 |
   3 | const baseUrl = 'https://www.saucedemo.com';
   4 |
   5 | test('user successfully login', async ({ page }) =>  {
   6 |     await page.goto(baseUrl);
   7 |
   8 |     await page.fill('#user-name', 'standard_user');
   9 |     await page.fill('#password', 'secret_sauce');
  10 |     await page.click('#login-button');
  11 |
  12 |     // Assert that the URL is correct after login
  13 |     await expect(page).toHaveURL(`${baseUrl}/inventory.html`);
  14 | });
  15 |
  16 | const usersLogin = [
  17 |     {
  18 |         username: '',
  19 |         password: '',
  20 |         error: 'Epic sadface: Username is required'
  21 |     },
  22 |     {
  23 |         username: '    '   ,
  24 |         password: '    ',
  25 |         error: 'Epic sadface: Username and password do not match any user in this service'
  26 |     },
  27 |     {
  28 |         username: 'wrong_user',
  29 |         password: '',
  30 |         error: 'Epic sadface: Password is required...'
  31 |     },
  32 |     {
  33 |         username: '',
  34 |         password: 'wrong_password',
  35 |         error: 'Epic sadface: Username is required'
  36 |     },
  37 |     {
  38 |         username: 'wrong_user',
  39 |         password: 'wrong_password',
  40 |         error: 'Epic sadface: Username and password do not match any user in this service'
  41 |     },
  42 |     {
  43 |         username: '~!@#$%^&*(',
  44 |         password: '~!@#$%^&*(',
  45 |         error: 'Epic sadface: Username and password do not match any user in this service'
  46 |
  47 |     },
  48 | ]
  49 |
  50 | usersLogin.forEach(({username, password, error}, index) => {
  51 |     test(`(${index}) user failed login using ${username}`, async ({ page }) =>  {
  52 |         await page.goto(baseUrl);
  53 |
  54 |         await page.fill('#user-name', username);
  55 |         await page.fill('#password', password);
  56 |         await page.click('#login-button');
  57 |
  58 |         // Assert error message
> 59 |         await expect(page.locator("[data-test='error']")).toHaveText(error);
     |                                                           ^ Error: Timed out 5000ms waiting for expect(locator).toHaveText(expected)
  60 |     });
  61 | });
  62 |
```